parquet/src/test/java/org/apache/iceberg/parquet/ParquetWritingTestUtils.java

```diff
@@ -53,20 +54,43 @@ static File writeRecords(TemporaryFolder temp, Schema schema,
   }
 
   static File writeRecords(
-      TemporaryFolder temp,
-      Schema schema, Map<String, String> properties,
-      Function<MessageType, ParquetValueWriter<?>> createWriterFunc,
-      GenericData.Record... records) throws IOException {
+          TemporaryFolder temp,
+          Schema schema, Map<String, String> properties,
+          Function<MessageType, ParquetValueWriter<?>> createWriterFunc,
+          GenericData.Record... records) throws IOException {
+    File file = createTempFile(temp);
+    write(file, schema, properties, createWriterFunc, records);
+    return file;
+  }
+
+  static long write(File file, Schema schema, Map<String, String> properties,
+                    Function<MessageType, ParquetValueWriter<?>> createWriterFunc,
+                    GenericData.Record... records) throws IOException {
+    FileAppender<GenericData.Record> writer = Parquet.write(localOutput(file))
```

Comment:
This should still use try-with-resources. You just have to keep a reference to the writer outside of it:

```java
  FileAppender<Record> writer = Parquet.write(...);
  try (Closeable toClose = writer) {
    writer.addAll(records);
  }

  writer.length();
```